server/src/main/java/org/opensearch/search/profile/query/QueryProfiler.java

```diff
@@ -51,15 +51,19 @@
  *
  * @opensearch.internal
  */
-public final class QueryProfiler extends AbstractProfiler<ContextualProfileBreakdown<QueryTimingType>, Query> {
+public class QueryProfiler extends AbstractProfiler<ContextualProfileBreakdown<QueryTimingType>, Query> {
 
     /**
      * The root Collector used in the search
      */
     private InternalProfileComponent collector;
 
-    public QueryProfiler(boolean concurrent) {
-        super(concurrent ? new ConcurrentQueryProfileTree() : new InternalQueryProfileTree());
+    public QueryProfiler() {
```

Comment:
Let's make it explicit:

```suggestion
    public QueryProfiler(InternalQueryProfileTree tree) {
    }
```

Reply:
It's addressed now.